website/pages/docs/k8s/upgrade/index.mdx

```diff
@@ -7,63 +7,303 @@ description: Upgrade Consul on Kubernetes
 `updatePartition` value are updated (zero-indexed). Therefore, by setting
 it equal to replicas, none should update yet.
 
-Next, run the upgrade. You should run this with `--dry-run` first to verify
-the changes that will be sent to the Kubernetes cluster.
+The `updateStrategy` controls how Kubernetes rolls out changes to the client daemonset.
+By setting it to `OnDelete`, no clients will be restarted until their pods are deleted.
+Without this, they would be redeployed alongside the servers because their Docker
+image versions have changed. This is not desirable because we want the Consul
+servers to be upgraded _before_ the clients.
 
-```shell-session
-$ helm upgrade consul ./
-...
+Next, run the upgrade.
+
+```bash
+helm upgrade consul hashicorp/consul --version <your-version> -f /path/to/my/values.yaml
 ```
 
-This should cause no changes (although the resource will be updated). If
+This will not cause the servers to redeploy (although the resource will be updated). If
 everything is stable, begin by decreasing the `updatePartition` value by one,
-and running `helm upgrade` again. This should cause the first Consul server
+and running `helm upgrade` again. This will cause the first Consul server
 to be stopped and restarted with the new image.
 
-Wait until the Consul server cluster is healthy again (30s to a few minutes)
-then decrease `updatePartition` and upgrade again. Continue until
+Wait until the Consul server cluster is healthy again (30s to a few minutes).
+This can be confirmed by running `consul members` on one of the previous servers
+and ensuring that all three servers are listed and are `alive`.
+
+Decrease `updatePartition` by one and upgrade again. Continue until
 `updatePartition` is `0`. At this point, you may remove the
 `updatePartition` configuration. Your server upgrade is complete.
 
 ## Upgrading Consul Clients
 
-With the servers upgraded, it is time to upgrade the clients. To upgrade
-the clients, set the `client.image` value to the desired Consul version.
+With the servers upgraded, it is time to upgrade the clients.
+If you are using Consul's service mesh features, you will want to be careful
+restarting the clients as outlined in [Service Mesh](#service-mesh).
+
+You can either:
+
+1. Manually run `kubectl delete pod <id>` for each consul daemonset pod
+2. Set the updateStrategy to rolling update with a small number:
+
+```values
+client:
+  updateStrategy: |
+    rollingUpdate:
+      maxUnavailable: 2
+    type: RollingUpdate
+```
+
 Then, run `helm upgrade`. This will upgrade the clients in batches, waiting
 until the clients come up healthy before continuing.
 
+3. Cordon and drain each node to ensure there are no connect pods running on it and then delete the
+   consul client pod on that node.
+
+If you are using only the Service Sync functionality, you can run an upgrade without
+being as careful since that component is more resilient to brief restarts of
```

Comment:
I'm not sure exactly how to re-word this but I think it's a bit vague.
```suggestion
requiring update strategies since that component is more resilient to brief restarts of
```